lib/mss/content_protection.js

```diff
@@ -0,0 +1,338 @@
+   * @return {!Array.<shaka.extern.DrmInfo>}
+   */
+  static parseFromProtection(elems, keySystemsBySystemId) {
+    const ContentProtection = shaka.mss.ContentProtection;
+    const XmlUtils = shaka.util.XmlUtils;
+
+    /** @type {!Array.<!Element>} */
+    let protectionHeader = [];
+    for (const elem of elems) {
+      protectionHeader = protectionHeader.concat(
+          XmlUtils.findChildren(elem, 'ProtectionHeader'));
+    }
+    if (!protectionHeader.length) {
+      return [];
+    }
+    return ContentProtection.convertElements_(
+        protectionHeader, keySystemsBySystemId);
+  }
+
+  /**
+   * Parses an Array buffer starting at byteOffset for PlayReady Object Records.
+   * Each PRO Record is preceded by its PlayReady Record type and length in
+   * bytes.
+   *
+   * PlayReady Object Record format: https://goo.gl/FTcu46
+   *
+   * @param {!DataView} view
+   * @param {number} byteOffset
+   * @return {!Array.<shaka.mss.ContentProtection.PlayReadyRecord>}
+   * @private
+   */
+  static parseMsProRecords_(view, byteOffset) {
+    const records = [];
+
+    while (byteOffset < view.byteLength - 1) {
+      const type = view.getUint16(byteOffset, true);
+      byteOffset += 2;
+
+      const byteLength = view.getUint16(byteOffset, true);
+      byteOffset += 2;
+
+      if ((byteLength & 1) != 0 || byteLength + byteOffset > view.byteLength) {
+        shaka.log.warning('Malformed MS PRO object');
+        return [];
+      }
+
+      const recordValue = shaka.util.BufferUtils.toUint8(
+          view, byteOffset, byteLength);
+      records.push({
+        type: type,
+        value: recordValue,
+      });
+
+      byteOffset += byteLength;
+    }
+
+    return records;
+  }
+
+  /**
+   * Parses a buffer for PlayReady Objects.  The data
+   * should contain a 32-bit integer indicating the length of
+   * the PRO in bytes.  Following that, a 16-bit integer for
+   * the number of PlayReady Object Records in the PRO.  Lastly,
+   * a byte array of the PRO Records themselves.
+   *
+   * PlayReady Object format: https://goo.gl/W8yAN4
+   *
+   * @param {BufferSource} data
+   * @return {!Array.<shaka.mss.ContentProtection.PlayReadyRecord>}
+   * @private
+   */
+  static parseMsPro_(data) {
+    let byteOffset = 0;
+    const view = shaka.util.BufferUtils.toDataView(data);
+
+    // First 4 bytes is the PRO length (DWORD)
+    const byteLength = view.getUint32(byteOffset, /* littleEndian= */ true);
+    byteOffset += 4;
+
+    if (byteLength != data.byteLength) {
+      // Malformed PRO
+      shaka.log.warning('PlayReady Object with invalid length encountered.');
+      return [];
+    }
+
+    // Skip PRO Record count (WORD)
+    byteOffset += 2;
+
+    // Rest of the data contains the PRO Records
+    const ContentProtection = shaka.mss.ContentProtection;
+    return ContentProtection.parseMsProRecords_(view, byteOffset);
+  }
+
+  /**
+   * PlayReady Header format: https://goo.gl/dBzxNA
+   *
+   * @param {!Element} xml
```

Comment:
This parameter should have a description saying what it is meant to be.
As-is, `getPlayReadyLicenseUrl` and `getLaurl_` have names that make them sound very similar. Saying that this parameter is meant to be the header of a PRO would help contextualize what this method does.

Reply:
Added a better description